src/rdf4cpp/storage/reference_node_storage/detail/BiDirFlatMap.hpp

```diff
@@ -174,13 +259,29 @@ struct BiDirFlatMap {
         return it->id;
     }
 
+    /**
+     * Look up the id (or ids, in case backwards is a multiset) corresponding the given view
+     *
+     * @param view view of value of which to find the id
+     * @return range of ids of the value if there were any, otherwise empty range if no id was found
+     */
+    [[nodiscard]] std::ranges::range auto lookup_id_range(view_type const &view) const noexcept {
```

Comment:
Ah, just one more thing: we should use https://en.cppreference.com/w/cpp/container/multimap to enable only the functions for the respective underlying map type.

Reply:
I don't think thats necessary. Multi and non-multi maps support both kinds of functions

Reply:
This was the wrong function to comment to. 

My main concern is that `lookup_id` is called when the `backward_type` is a multi-map. This would be most probably by mistake and should not be allowed. Calling lookup_id_range on a non-multi-map is ok but doesn't make much sense.

Reply:
I somewhat agree with your concern, but I don't think this is a good idea, because:

- `multimap` itself also has `find`, so there must be some usecase where you may want to use that.

- trying to detect container behaviour based on an interface is inherently flawed and does not work in C++. Just because an object has a specific interface does not mean that it behaves a certain way. Constraints do not have the purpose to know what an object does, just that it can do it (mainly to improve error messages). In contrast, rust traits determine what an object does, not just that they can do it. So we would be disabling functionality based on a guess what the behaviour might be.

Reply:
ok, good points. Then let's keep things as they are.